### ISSUE TITLE
<a> tag inside web-component reload the page

### ISSUE THREAD

**Issue by FrancoisDF** [sveltech/svelte-filerouter#451]
**bug description**
When using a web-component with a `<a>` tag, routify reloads the app. 

**Step to reproduce**
From one of the pages index.svelte, add the following code
```html
<link-about></link-about>

<script>
  customElements.define("link-about", class extends HTMLElement {
      constructor() {
      const innerHTML = "<style>:host{color:red}</style><a href='/about'>about</a>";
      super()
          .attachShadow({mode: "open"})
          .innerHTML = innerHTML;
      }
  })
</script>
```
Compare to a regular `<a>` tag, the shadow element is not behaving as expected.

**version**
Tested with routify@2 and routify@3

**Comment by FrancoisDF** [comment]
https://github.com/roxiness/routify/blob/a1e5db7151fa654566ef472d15407d49ff38b480/runtime/navigator.js#L102
This is actually due to the `event.target.closest('a')` code that return events coming from the root custom element. To allow both regular and shadowed `<a>`  tag to work, we would need to use `event.target.closest('a') || event.composedPath()[0]`. 
```js
  const el = event.target.closest('a') || event.composedPath()[0]
```

**Comment by jakobrosenberg** [comment]
This looks good to me. I admittedly don't know how `event.composedPath()` works. Are there any drawbacks?

**Comment by FrancoisDF** [comment]
It's a possible replacement of `event.target` that array out the path of the event through the DOM structure instead of returning the DOM. The biggest difference is that it works on custom element with open shadow (`event.target` return the root custom element). It will be automatically initialized with at least one element as I understood.
The current implementation only focus on the first element being clicked by the user. Maybe a bit less generic than a "closet('a')" implementation, but every test I did would anyway target the `a` tag first. Even behind a custom element.
The only drawback I see is that it's not compatible with IE. We could add something like `event.composedPath &&  event.composedPath()[0]`, but I'm not sure that is necessary. 
Let me know if you want me to open a PR or if you can take care of it. Thanks for considering it anyway!

**Comment by FrancoisDF** [comment]
Do you think we can get this pushed to a new v2 version? Do you want me to open a PR from the master branch?

**Comment by ghostdevv** [comment]
LGTM - although I don't know what side effects it might have so might be good to put it behind a option/

**Comment by jakobrosenberg** [comment]
I'd be interested in a PR. Especially for R3, but R2 works too.

Should we use something like this?

```javascript
  const isAnchorElem = (el) => /** determine if element is anchor */
  const el = event.target.closest('a') || event.composedPath().find(isAnchorElem)
```

**Comment by FrancoisDF** [comment]
That's a good idea considering that it will behave the same as the `closest()` function does but through an array of elements.
After more testing, if the `a` tag had a deep structure, then the first element of the list will not be the anchor. for example with:
```html
<script>
  customElements.define("link-about", class extends HTMLElement {
      constructor() {
      const innerHTML = "<a href='/about'> <h3> about </h3> </a>";
      super()
          .attachShadow({mode: "open"})
          .innerHTML = innerHTML;
      }
  })
</script>
```
`composedPath()` will return `[h3,a,...]`

So considering that, using the `find` function will better support such situations (as `closest` does by default).

```
const el = event.target.closest('a') || event.composedPath().find(el => el.tagName === 'A')
```

I will open a PR against master that we can probably merge to R2 and R3 once validated.

**Comment by FrancoisDF** [comment]
I'm not sure about the process, but should we expect this fix to be published to V2.18.6 anytime soon? Thanks!

**Comment by ghostdevv** [comment]
Just realised your commit message wasn't [a conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) so the semantic release bot wouldn't have picked it up, will try to fix this
